### ISSUE TITLE
Document Build Process

### ISSUE THREAD

**Issue by joshuaflanagan** [TeamShinkansen/Hakchi2-CE#146]
I would appreciate some documentation (in the readme or the wiki, or at the very least, a comment on this issue) about how the developers work with this source code and build the results. I have attempted to build it using Visual Studio 2017 on Windows 10, and it fails on the PreBuildEvent's use of `busybox` (I think Windows wants it referred to as `busybox.exe`. When I clear out the PreBuildEvent, the solution compiles successfully. Since I assume the PreBuildEvents work for the developers on the project, there must be something different about their environment.

1) Do you build on Windows, linux, or OSX? If on Windows, are you relying on something like msys?
2) Are you using Visual Studio (which version?), MonoDevelop, or command line tooling?

Any more details would be appreciated.

**Comment by daPhie79** [comment]
you're right, it probably needs some guidance. we've just been a small team, and we don't have lots of time to work on it collectively.

we've tried to make it work on monodevelop, and while i managed to compile it in ubuntu 18.04, it wasn't working at all, so i didn't commit all the changes needed to do so (busybox without the .exe was one of these changes)

as for windows, our main dev env, it does work, but you need git installed in your path. i think that's what was causing me the busybox error back when i reinstalled windows 10 from scratch. try this first.

**Comment by KMFDManic** [comment]
I've built various Hakchis several times on Visual Studio. I have used the most recent one for free. Typically, I would make sure to install anything and everything that would constitute a dependency, then adjust accordingly for any compile errors.  Been doing the compiles with Windows 7.  How are things going, compile wise, now?

**Comment by pathartl** [comment]
I've been talking with @DanTheMan827 about this on the side. I think the current plan is to evaluate something like Vagrant to provide a unified dev environment.
